### PR TITLE
KAFKA-3865: Fix transient failure in WorkerSourceTaskTest.testSlowTaskStart

### PR DESCRIPTION
**Pull request by hachikuji**

### CODE REVIEW

**Comment by hachikuji** [comment]
Ping @ewencp

**Comment by Ishiihara** [comment]
Making the changes to synchronous is less sensitive to timing. I think It should be fine as unit tests are checking the correctness for basic cases.

**Comment by Ishiihara** [comment]
As discussed offline with @hachikuji, the addition of `future.get()` is to ensure that the exception thrown by the executor is propagated.  Also it is not safe to throw the `InterruptedException` as the flag set by the exception is only cleared with next `awaitLatch`.

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java [hunk 1]
```diff
@@ -68,7 +65,6 @@
 
 @RunWith(PowerMockRunner.class)
 public class WorkerSourceTaskTest extends ThreadedTest {
```
Comment (ewencp) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:67]: Can we just drop the `ThreadedTest` now? We no longer have a `ShutdownableThread` which is what `ThreadedTest` relies on.

Comment (ewencp) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:67]: Actually, there are a few other tests that use this too. I'll file a JIRA to clean it up -- it's not reliable enough anyway (and maybe some of the tests can just drop the use of the thread and mock the necessary bits instead if that doesn't make things even messier).

Comment (hachikuji) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java:67]: We could probably add something which provides the same capability for an executor service though. That would be better than sprinkling `future.get()` calls all over the place.

**Comment by ewencp** [comment]
LGTM. Will merge to both trunk and 0.10.0 since it just cleans up the unit test and should hopefully help w/ stability of 0.10.0 builds too.

**Pull request closed without merging**